### ISSUE TITLE
ONNX support

### ISSUE THREAD

**Issue by quaertym** [denizyuret/Knet.jl#302]
From the ONNX(Open Neural Net Exchange) website:

> ONNX is an open format to represent deep learning models. With ONNX, AI developers can more easily move models between state-of-the-art tools and choose the combination that is best for them. 

I think import/export functionality for ONNX format is a valuable feature. It has the potential to accelerate model transfer from other implementations. It will lower the barrier to entry and encourage many to try Knet. With this, we will have a standard way to share models. Once trained, the model can be run on different architectures (mobile for instance).

It is supported by industry (Amazon, Facebook, Microsoft, Nvidia, Intel, IBM and many more) and already adopted by major DL frameworks(PyTorch, Caffe2, Chainer and more).

More on ONNX: [Website](https://onnx.ai/), [GitHub](https://github.com/onnx/onnx)

**Comment by rened** [comment]
@ayush1999 is currently working on an ONNX importer, as a GSoC project: https://github.com/FluxML/ONNX.jl

Perhaps some code can be shared. It would be also great to have export capabilies models, in addition to import.

**Comment by ayush1999** [comment]
Yeah, sounds great. My project mainly focuses on importing ONNX models into the Flux framework. However, my mentor @PhilTomson was interested in making a verilog backend, so that we could import these models to various other frameworks. This could be a good starting point.

**Comment by rened** [comment]
@ayush1999 What do you mean by "verilog backend"? I only know "verilog" as the  [hardware description language](https://en.wikipedia.org/wiki/Verilog), but I don't think that is what you are referring to in this context?

**Comment by ayush1999** [comment]
By verilog, I mean a common backend from which we could export model to different frameworks.

**Comment by ayush1999** [comment]
@rened  @quaertym  Are you still interested in this? Most of my work in ONNX.jl for loading models into Flux is complete, and I'm able to load models with a pretty good accuracy. The only difference between implementing ONNX reader for Flux and Knet would be the graph.jl and ops.jl files, since they vary from framework to framework. Ping me if you're interested.

**Comment by denizyuret** [comment]
Definitely interested.

**Comment by denizyuret** [comment]
Ayush, please contact me at denizyuret@gmail.com

**Comment by ayush1999** [comment]
There is an initial repo up [here](https://www.github.com/ayush1999/ONNXKnet.jl).
 @denizyuret would be great if you could check your mail. Thanks!

**Comment by ayush-1506** [comment]
@denizyuret Sorry for the confusion: I lost access to my previous github account(@ayush1999 ). This is my new account. Could you send me the invite to this account? Thanks.

**Comment by denizyuret** [comment]
@ayush1999 repo https://github.com/ayush1999/ONNXKnet.jl was copied to KnetML: https://github.com/KnetML/ONNXKnet.jl.
@egeersu started another https://github.com/egeersu/KnetONNX.jl.
I think I will integrate these into the main Knet package. Use AutoGrad computational graph for export.
